apis/leds/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+#![no_std]
+
+use libtock_platform::Syscalls;
+
+/// The LEDs driver
+///
+/// # Example
+/// ```ignore
+/// use libtock2::Leds;
+///
+/// // Turn on led 0
+/// Leds::on(0);
+/// ```
+pub struct Leds<S: Syscalls>(S);
+
+impl<S: Syscalls> Leds<S> {
+    /// Run a check against the leds capsule to ensure it is present.
+    ///
+    /// Returns `true` if the driver was present. This does not necessarily mean
+    /// that the driver is working, as it may still fail to allocate grant
+    /// memory.
+    #[inline(always)]
+    pub fn driver_check() -> bool {
+        S::command(DRIVER_ID, DRIVER_CHECK, 0, 0).is_success_u32()
+    }
+
+    #[inline(always)]
+    pub fn count() -> usize {
+        S::command(DRIVER_ID, LEDS_COUNT, 0, 0)
+            .get_success_u32()
+            .unwrap_or_default() as usize
+    }
+
+    #[inline(always)]
+    pub fn on(led: usize) {
```

Comment:
The indices should be `u32`s, not `usize`. We can revisit this if Tock gains a 64-bit ABI, but until then `libtock_platform` is using `u32` as its typical numeric type.